lizrd/support/logging.py

```diff
@@ -251,6 +252,77 @@ def flush_if_necessary(self):
         pass
 
 
+class WandbLogger(AbstractLogger):
+    def __init__(self, logger, args: Namespace):
+        super().__init__(logger, args)
+        self.random_id = generate_random_string(8)
+
+    def _make_path(self, title: str, series: Optional[str] = None):
+        parts = [title]
+        if series is not None:
+            parts.append(series)
+        return "/".join(parts)
+
+    def report_generic_info(self, *, title: str, iteration: int, data):
+        if isinstance(data, plotly.graph_objs.Figure):
+            self.report_plotly(figure=data, title=title, iteration=iteration)
+        elif isinstance(data, list):
+            if isinstance(data[0], float):
+                for i, scalar in enumerate(data):
+                    self.report_scalar(
+                        title=title, value=scalar, series=str(i), iteration=iteration
+                    )
+            else:
+                raise NotImplementedError()
+        else:
+            self.report_scalar(title=title, value=data, iteration=iteration)
+
+    def report_scalar(
+        self,
+        *,
+        title: str,
+        value: float,
+        iteration: int,
+        series: Optional[str] = None,
+    ):
+        path = self._make_path(title, series)
+        assert (not math.isnan(value)) and (
+            not math.isinf(value)
+        ), f"Trying to log {path} as {value}. Neptune doesn't allow logging NaN or Inf."
```

Comment:
Wandb

Reply:
Now we log inf / nan (Wandb allows that)